onnxruntime/core/session/inference_session.cc

```diff
@@ -102,12 +102,14 @@ InferenceSession::InferenceSession(const SessionOptions& session_options,
     : session_options_(session_options),
       graph_transformation_mgr_(session_options.max_num_graph_transformation_steps),
       logging_manager_(logging_manager),
+#ifdef USE_OPENMP
       thread_pool_(concurrency::CreateThreadPool("intra_op_thread_pool",
                                                  session_options.intra_op_num_threads)),
       inter_op_thread_pool_(session_options.execution_mode == ExecutionMode::ORT_PARALLEL
```

Comment:
>inter_op_thread_pool_ [](start = 6, length = 21)

Seems this would disable parallel executor when building OpenMP? Say, if running on a 8-HT machine, with OMP_NUM_THREADS=4 and -y 2 in onnxruntime_perf_test, is that a valid setting?

Reply:
This is a good point. Do you think it's better to keep the previous behavior for inter threadpool (-y) ?

Reply:
Parallel executor is off by default so it should not cause any perf difference in your case. By keeping this logic, we can try to enable it together with OpenMP in case some models might benefit from the combination.

---
In reply to: [350977021](https://github.com/microsoft/onnxruntime/pull/2485#discussion_r350977021) [](ancestors = 350977021)

Reply:
If USE_OPENMP is not defined, please initialize the var to nullptr here. Because the next few lines are referencing this variable.